src/server/tran_server.hpp

```diff
@@ -72,21 +72,62 @@ class tran_server
 
     int boot (const char *db_name);
 
-    void disconnect_page_server ();
-    bool is_page_server_connected () const;
-    bool is_page_server_connected (const size_t idx) const;
-    void push_request_to (size_t idx, tran_to_page_request reqid, std::string &&payload);
+    /* send request to a specific connection */
+    void push_request (size_t idx, tran_to_page_request reqid, std::string &&payload);
+    int send_receive (size_t idx, tran_to_page_request reqid, std::string &&payload_in, std::string &payload_out) const;
+
+    /* send request to the main connection */
     void push_request (tran_to_page_request reqid, std::string &&payload);
     int send_receive (tran_to_page_request reqid, std::string &&payload_in, std::string &payload_out) const;
 
+    void disconnect_all_page_servers ();
+    bool is_page_server_connected () const;
     virtual bool uses_remote_storage () const;
 
   protected:
-    using page_server_conn_t = cubcomm::request_sync_client_server<tran_to_page_request, page_to_tran_request, std::string>;
-    using request_handlers_map_t = std::map<page_to_tran_request, page_server_conn_t::incoming_request_handler_t>;
+    class connection_handler
```

Comment:
Nit.
I would take the `connection_handler` hierarchy outside the `*tran_server` hierarchy.
Your choice.

Reply:
hmm, do you think it's more readable? Is it the reason? or it could make developers make mistakes?